File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/AttributeSetInstanceExtensionFieldMvoStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class AttributeSetInstanceExtensionFieldMvoStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "attributeSetInstanceExtensionFieldIdGroupId",
            "attributeSetInstanceExtensionFieldIdIndex",
            "attrSetInstEFGroupVersion",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private AttributeSetInstanceExtensionFieldMvoStateEventId value;

    public AttributeSetInstanceExtensionFieldMvoStateEventIdFlattenedDto()
    {
        this(new AttributeSetInstanceExtensionFieldMvoStateEventId());
    }

    public AttributeSetInstanceExtensionFieldMvoStateEventIdFlattenedDto(AttributeSetInstanceExtensionFieldMvoStateEventId value)
    {
        this.value = value;
    }

    public AttributeSetInstanceExtensionFieldMvoStateEventId toAttributeSetInstanceExtensionFieldMvoStateEventId()
    {
        return this.value;
    }


    public String getAttributeSetInstanceExtensionFieldIdGroupId()
    {
        return this.value.getAttributeSetInstanceExtensionFieldId().getGroupId();
    }

    public void setAttributeSetInstanceExtensionFieldIdGroupId(String attributeSetInstanceExtensionFieldIdGroupId)
    {
        this.value.getAttributeSetInstanceExtensionFieldId().setGroupId(attributeSetInstanceExtensionFieldIdGroupId);
    }

    public String getAttributeSetInstanceExtensionFieldIdIndex()
    {
        return this.value.getAttributeSetInstanceExtensionFieldId().getIndex();
    }

    public void setAttributeSetInstanceExtensionFieldIdIndex(String attributeSetInstanceExtensionFieldIdIndex)
    {
        this.value.getAttributeSetInstanceExtensionFieldId().setIndex(attributeSetInstanceExtensionFieldIdIndex);
    }

    public Long getAttrSetInstEFGroupVersion()
    {
        return this.value.getAttrSetInstEFGroupVersion();
    }

    public void setAttrSetInstEFGroupVersion(Long attrSetInstEFGroupVersion)
    {
        this.value.setAttrSetInstEFGroupVersion(attrSetInstEFGroupVersion);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        AttributeSetInstanceExtensionFieldMvoStateEventIdFlattenedDto other = (AttributeSetInstanceExtensionFieldMvoStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/AttributeSetInstanceExtensionFieldMvoStateEventIdFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class AttributeSetInstanceExtensionFieldMvoStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'attributeSetInstanceExtensionFieldIdGroupId' => 'string',
            'attributeSetInstanceExtensionFieldIdIndex' => 'string',
            'attrSetInstEFGroupVersion' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var AttributeSetInstanceExtensionFieldMvoStateEventId
     */
    private $value;

    /**
     * @param AttributeSetInstanceExtensionFieldMvoStateEventId $value
     */
    public function __construct(AttributeSetInstanceExtensionFieldMvoStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new AttributeSetInstanceExtensionFieldMvoStateEventId();
        }
    }

    /**
     * @return AttributeSetInstanceExtensionFieldMvoStateEventId
     */
    public function toAttributeSetInstanceExtensionFieldMvoStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getAttributeSetInstanceExtensionFieldIdGroupId()
    {
        return $this->value->getAttributeSetInstanceExtensionFieldId()->getGroupId();
    }

    /**
     * @param string $attributeSetInstanceExtensionFieldIdGroupId
     */
    public function setAttributeSetInstanceExtensionFieldIdGroupId($attributeSetInstanceExtensionFieldIdGroupId)
    {
        $this->value->getAttributeSetInstanceExtensionFieldId()->setGroupId($attributeSetInstanceExtensionFieldIdGroupId);
    }

    /**
     * @return string
     */
    public function getAttributeSetInstanceExtensionFieldIdIndex()
    {
        return $this->value->getAttributeSetInstanceExtensionFieldId()->getIndex();
    }

    /**
     * @param string $attributeSetInstanceExtensionFieldIdIndex
     */
    public function setAttributeSetInstanceExtensionFieldIdIndex($attributeSetInstanceExtensionFieldIdIndex)
    {
        $this->value->getAttributeSetInstanceExtensionFieldId()->setIndex($attributeSetInstanceExtensionFieldIdIndex);
    }

    /**
     * @return Long
     */
    public function getAttrSetInstEFGroupVersion()
    {
        return $this->value->getAttrSetInstEFGroupVersion();
    }

    /**
     * @param Long $attrSetInstEFGroupVersion
     */
    public function setAttrSetInstEFGroupVersion($attrSetInstEFGroupVersion)
    {
        $this->value->setAttrSetInstEFGroupVersion($attrSetInstEFGroupVersion);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getAttributeSetInstanceExtensionFieldIdGroupId(),
            $this->getAttributeSetInstanceExtensionFieldIdIndex(),
            $this->getAttrSetInstEFGroupVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return AttributeSetInstanceExtensionFieldMvoStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setAttributeSetInstanceExtensionFieldIdGroupId($pValues[0]);
        $this->setAttributeSetInstanceExtensionFieldIdIndex($pValues[1]);
        $this->setAttrSetInstEFGroupVersion($pValues[2]);
        return $this;
    }

}

